nil/internal/collate/proposer.go

```diff
@@ -191,7 +191,9 @@ func (p *proposer) handleL1Attributes(tx db.RoTx, mainShardHash common.Hash) err
 		}
 	}
 
-	txn, err := CreateL1BlockUpdateTransaction(block)
+	txId := p.executionState.InTxCounts[types.MainShardId]
+	p.executionState.InTxCounts[types.MainShardId] = txId + 1
```

Comment:
Wouldn't we increment the InTxCounts twice? Once here, and once in AcceptInternalTransaction?

Reply:
No. Proposer and validator create different execution states

Reply:
Shouldn't it be `txId := p.executionState.InTxCounts[types.MainShardId]` instead?

Reply:
Right now it will work. But in case if we will have more system contracts in future we should increment id. Otherwise all transaction will have the same TxId. That will produce TxGap error

Reply:
I guess we'll migrate to contract-based messaging before that happens (hopefully)